src/framework/audio/internal/worker/playback.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "playback.h"
+
+#include "log.h"
+#include "async/async.h"
+
+#include "internal/audiothread.h"
+#include "internal/audiosanitizer.h"
+
+#include "internal/worker/playerhandler.h"
+#include "internal/worker/trackshandler.h"
+#include "internal/worker/audiooutputhandler.h"
+#include "internal/worker/tracksequence.h"
+
+#include "audioerrors.h"
+
+using namespace mu;
+using namespace mu::audio;
+using namespace mu::async;
+
+Playback::Playback()
+{
+    m_playerHandlersPtr = std::make_shared<PlayerHandler>(this);
+    m_trackHandlersPtr = std::make_shared<TracksHandler>(this);
+    m_audioOutputPtr = std::make_shared<AudioOutputHandler>(this);
+}
+
+Promise<TrackSequenceId> Playback::addSequence()
+{
+    return Promise<TrackSequenceId>([this](Promise<TrackSequenceId>::Resolve resolve, Promise<TrackSequenceId>::Reject /*reject*/) {
+        TrackSequenceId newId = m_sequences.size();
```

Comment:
I think it better to add check tread marker

Reply:
Ah, completely forgot this one, thanks